File: mobile/flutter/lib_flutter_base/android/src/main/java/com/codesdancing/flutter/STFlutterInitializer.kt
package com.codesdancing.flutter

import android.app.Application
import com.codesdancing.flutter.boost.STFlutterBoostInitializer
import com.codesdancing.flutter.multiple.STFlutterMultipleInitializer


@Suppress("unused")
object STFlutterInitializer {

    const val TAG = "[flutter]"

    @JvmStatic
    var application: Application? = null
        private set

    @JvmStatic
    var enableMultiple: Boolean = true
        private set

    @JvmStatic
    var debug: Boolean = false
        private set

    private var isInitialized: Boolean = false

    @JvmStatic
    fun initial(application: Application?, debug: Boolean = STFlutterInitializer.debug) {
        if (isInitialized) return
        application ?: return

        this.application = application
        this.debug = debug

        if (enableMultiple) {
            STFlutterMultipleInitializer.initial(application, debug)
        } else {
            STFlutterBoostInitializer.initial(application, debug)
        }
        
        isInitialized = true
    }
}

File: mobile/flutter/lib_flutter_base/ios/Classes/FlutterBase/STFlutterInitializer.m
//
//  STFlutterInitializer.m
//  lib_flutter_base
//
//  Created by krmao on 2021/3/9.
//

#import "STFlutterInitializer.h"
#import "STFlutterBoostInitializer.h"
#import "STFlutterMultipleInitializer.h"

static STFlutterInitializer *instance = nil;

@implementation STFlutterInitializer

+(instancetype)sharedInstance{
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        instance = [[STFlutterInitializer alloc] init];
    });
    return instance;
}

-(void) initial:(UIApplication*)application debug:(BOOL)debug{
    NSLog(@"STFlutterInitializer initial application=%@ _isInitialized=%d", application, _isInitialized);
    
    if(_isInitialized == YES){
        return;
    }
    _debug = debug;
    _application = application;
    _enableMultiple = YES;
    
    if(_enableMultiple){
        [[STFlutterMultipleInitializer sharedInstance] initial:application debug:debug];
    }else{
        [[STFlutterBoostInitializer sharedInstance] initial:application debug:debug];
    }
    
    _isInitialized = YES;
}

@end
